pandas/core/indexes/base.py

```diff
@@ -1006,7 +1006,7 @@ def ravel(self, order: str_t = "C") -> Self:
         """
         return self[:]
 
-    def view(self, cls=None):
+    def view(self, cls=None) -> np.ndarray:
```

Comment:
this is often going to return an Index.  if anything we might want to deprecate the paths that dont

Reply:
Yeah agree, Self was correct, the other parts are buggy